### DISCUSSION TITLE
Playground text completion queries to /v1/chat/completions instead of /v1/completions?

### DISCUSSION THREAD

**Discussion by Chris-Tran-2023** [open-webui/open-webui#4244] [General]
I am using Aphrodite-engine for hosting Mixtral-8x7b using exllamav2 quantization. I have connected to it via the OpenAI API connection in the settings. I was planning to use Open-WebUI for serving chat to multiple users, but also wanted to keep the completions/generation, like in text-generation-webUI's notebook mode. I thought the Playground should function similar, using the text completion switch. My expectation is this should query to /v1/completions, rather than /v1/chat/completions, but my log in aphrodite shows POST ot /v1/chat/completions. Is this intended behavior? Screenshots below

![image](https://github.com/user-attachments/assets/db52aa6f-2010-438a-8221-9df220e75bec)

![image](https://github.com/user-attachments/assets/4953d095-4bef-4c33-851a-eef36eff9945)

**Comment by cduk** [comment]
Yes, it is calling the wrong endpoint.

**Reply by justinh-rahb** [reply]
No, it's not. `/v1/completions` is legacy.

**Reply by cduk** [reply]
It may be legacy, but it is still in use. Is there a way to use base models (non-instruct/non-chat) via the chat completions endpoint? vLLM requires now a chat template for the chat endpoint. I'm not sure if/how one can create a dummy chat completions endpoint that would allow for the use of base models with the playground that would generate the same output as when using the completions endpoint with the base model.

I'm not sure how you would use models such as gpt-3.5-turbo-instruct. Unless it is intended to drop support for these older models and all base models - or maybe there is a workaround with chat templates.

It seems strange that playground has only chat/completions esp. when the chat dropdown is deselected and completions is selected instead:

![image](https://github.com/user-attachments/assets/2d062f03-f520-428e-9c2d-df469101de91)

But right now, as-is, the text completion mode is broken.
